### PR TITLE
[docs-infra] Remove randomized API page layout

### PR DESCRIPTION
**Pull request by alexfauquette**
From https://github.com/mui/material-ui/pull/40862 Allows data grid API pages to use list by default thanks to the `defaultLayout` props

### CODE REVIEW

File: docs/src/modules/utils/layoutStorageKey.tsx [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export const dataGridLayoutKeys = {
+  slots: 'dataGrid_apiPage_slots',
```
Comment (cherniavskii) [inline docs/src/modules/utils/layoutStorageKey.tsx:2]: Side note: I just realized that the slots section is missing the v7 data grid docs:

v6: https://mui.com/x/api/data-grid/data-grid/#slots
v7: https://next.mui.com/x/api/data-grid/data-grid/#slots

Any ideas on how to bring it back? 😅

**Review by cherniavskii — APPROVED** [review]
Thanks!

File: docs/src/modules/utils/layoutStorageKey.tsx (hunk 1 above)
Comment (alexfauquette) [reply docs/src/modules/utils/layoutStorageKey.tsx:2]: Yes, the slots are generated by detecting interfaces named `[componentName]Slots`.

You can override this behavior with `slotInterfaceName`

Here is the uscase for date pickers:
https://github.com/mui/mui-x/blob/master/scripts/buildApiDocs/pickersSettings/getComponentInfo.ts/#L26-L28

Comment (alexfauquette) [reply docs/src/modules/utils/layoutStorageKey.tsx]: I added the fix

**Pull request merged**